Release v0.8.0

smaye81 opened the pull request:
We have reorganized the repository a bit recently to allow for better integration of Connect for Node.js.  The repo is now named **Connect-ES** and contains packages for working with Connect from browser clients and for working with servers and clients in Node.js.  As part of the reorganization, this release includes numerous changes, some of which are breaking depending on which packages are being used.  Detailed release notes for each are as follows:

## connect-web

All imports from `@bufbuild/connect-web` still work, but we moved core types to the `@bufbuild/connect` package. Please add `@bufbuild/connect` to your dependencies when you update, and change your imports of these types to `@bufbuild/connect` at your leisure.  Please note, though, they will be removed in one of the next releases.

### Enhancements
* Allow JSON in the gRPC-web transport of `@bufbuild/connect-web` by @timostamm in #334.

### Fixes
*  Emit unpadded base64 for binary headers by @timostamm in #454
*  Better errors caused by non-compliant response body by @timostamm (fixed as part of #474).

## protoc-gen-connect-web

The code generator plugin `@bufbuild/protoc-gen-connect-web` has been renamed to `@bufbuild/protoc-gen-connect-es`.  The old plugin is still being published to npm, but we encourage users to migrate to the new one at their earliest convenience.  `@bufbuild/protoc-gen-connect-web` will be deprecated soon and we will no longer make changes to that package.  

To migrate to the new plugin, see this [README](https://github.com/bufbuild/connect-es/tree/main/packages/protoc-gen-connect-web) for details.

## connect-node

### 🚨 Breaking Changes 🚨 

We apologize, but as part of our omelet making, we had to break some eggs.  As part of this upgrade, you will need to switch to use the new `ConnectRouter`.  This is a _much_ more intuitive and flexible approach to implementing your endpoints for various frameworks, so we hope the benefits outweigh the effort to migrate.

Please see [these docs](https://github.com/bufbuild/connect-es/tree/main/packages/connect-node#connectnodeadapter) for steps on switching to `ConnectRouter`.

In addition, once you update to v0.8.0 of `@bufbuild/connect-node` you will need to do the following:

* Remove any dependencies on `@bufbuild/connect-core`.
* Add a dependency on `@bufbuild/connect`.

Finally, we encourage you to try [Fastify](https://www.fastify.io/).  We now offer support via our new [`@bufbuild/connect-fastify`](https://www.npmjs.com/package/@bufbuild/connect-fastify) plugin.  ([source](https://github.com/bufbuild/connect-es/tree/main/packages/connect-fastify)).

### Enhancements
*  Add support for Fastify by @timostamm in #474
*  Add support for Express by @timostamm in #479
*  Fastify and Express support made courtesy of `ConnectRouter` by @timostamm in #473
  
### Fixes
*  Move polyfill dep to `connect-node` by @smaye81 in #462

## Other changes
* NodeNext module resolution by @fubhy in #485

fubhy commented:
Hey Steve! Can you include this (https://github.com/bufbuild/connect-es/pull/485) before publishing this? :-)

timostamm commented:
@smaye81, let's discuss offline how we can provide good guidance for updating.

smaye81 commented:
Note:  Browserstack is having issues, preventing those tests from running.  But, there were no changes since the last successful run there, so merging is safe.
